components/api/core/api_impl.bal

```diff
@@ -491,3 +492,74 @@ returns http:Response {
         return buildUnknownErrorResponse();
     }
 }
+
+# Update an existing image
+#
+# + updateImageReq - received request which contains header
+# + orgName - Organization name
+# + imageName - Image ID
+# + return - http response (200 if success, 405 or 500 otherwise)
+public function updateImage (http:Request updateImageReq, string orgName, string imageName, gen:ImageUpdateRequest updateImageBody) returns http:Response {
+    if (updateImageReq.hasHeader(constants:AUTHENTICATED_USER)) {
+        string userId = updateImageReq.getHeader(constants:AUTHENTICATED_USER);
+        log:printDebug(io:sprintf("%s is attempting to update image \'%s\' in organization %s", userId, imageName, orgName));
+        log:printDebug(updateImageBody.description);
+        sql:UpdateResult | error? updateImageRes = db:updateImage(orgName, imageName, updateImageBody.description, userId);
```

Comment:
Is description mandatory ? Otherwise this might fail if the description is not sent from outside.